GUI/src/cz/fidentis/gui/comparison_batch/BatchComparisonConfiguration.java

```diff
@@ -444,110 +437,104 @@ private void processComparisonButtonActionPerformed(java.awt.event.ActionEvent e
     }//GEN-LAST:event_jButton2ActionPerformed
 
     private void jButton10ActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_jButton10ActionPerformed
         final ProjectTopComponent tc = GUIController.getSelectedProjectTopComponent();
-        ResultExports.instance().saveRegisteredModelsBatch(tc, tc.getProject().getSelectedBatchComparison().getRegistrationResults(),
-                tc.getProject().getSelectedBatchComparison().getModels(), "_batch");
+        ResultExports.instance().saveRegisteredModelsBatch(tc, getContext().getRegistrationResults(),
+                getContext().getModels(), "_batch");
     }//GEN-LAST:event_jButton10ActionPerformed
 
     private void exportLandmarksButtonActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_exportLandmarksButtonActionPerformed
         final ProjectTopComponent tc = GUIController.getSelectedProjectTopComponent();
-        FPImportExport.instance().exportBatch(tc, tc.getProject().getSelectedBatchComparison());
+        FPImportExport.instance().exportBatch(tc, getContext());
     }//GEN-LAST:event_exportLandmarksButtonActionPerformed
 
     public Boolean getScaleEnabled() {
-        return jCheckBox2.isSelected();
+        return fpScaleCheckBox.isSelected();
     }
 
     public void setProcessComparisonEnabled(boolean en) {
         processComparisonButton.setEnabled(en);
-        GUIController.getSelectedProjectTopComponent().getProject().getSelectedBatchComparison().setCompareButtonEnabled(en);
+        getContext().setCompareButtonEnabled(en);
     }
 
-    private void setupComparisonMethods(BatchComparison c) {
-        jComboBox1.removeAllItems();
-        for (int i = 0; i < ComparisonMethod.values().length; i++) {
-            jComboBox1.addItem(ComparisonMethod.values()[i]);
-            if (c.getComparisonMethod() == ComparisonMethod.values()[i]) {
-                jComboBox1.setSelectedIndex(i);
-            }
-        }
-        /*  if((c.getRegistrationMethod() == 0)){
-            jComboBox1.removeItemAt(0);
-        }*/
-        if ((c.getRegistrationMethod() != RegistrationMethod.PROCRUSTES)) {
-            jComboBox1.removeItemAt(1);
-        }
-
+    private BatchComparison getContext(){
+        return GUIController.getSelectedProjectTopComponent().getProject().getSelectedBatchComparison();
     }
-
+    
     public void setConfiguration() {
-        BatchComparison c = GUIController.getSelectedProjectTopComponent().getProject().getSelectedBatchComparison();
+        BatchComparison c = getContext();
+        RegistrationMethod reg = c.getRegistrationMethod();
+        
+        //because some items might need to be removed based on the mode later
+        comparisonComboBox.setModel(new DefaultComboBoxModel(ComparisonMethod.values()));
+
+        
+        comparisonComboBox.setSelectedItem(c.getComparisonMethod());
+        fpScaleCheckBox.setSelected(c.isFpScaling());
+        jButton2.setVisible(reg == RegistrationMethod.HAUSDORFF);
```

Comment:
asi by bolo fajn premenovat `jButton2` na nieco zrozumitelne. Inak k tomuto este - toto su asi veci kde treba ukazat/schovat niektore casti GUI na zaklade toho co je vybrate v modeli. V takomto pripade som to ja riesil tak, ze najskor som nastavil spravnu hodnotu do daneho inputu na ktorom zalezi co sa ma schovat/ukazat a potom som zavolal jeho `onAction` metodu ktora sa mi postara klasicky o to ze sa vyriesi co schovat a co ukazat. Takyto postup je mozno trochu divny :D ale predides tak tomu ze musis rovnaku logiku duplikovat tu. Dufam ze sa da pochopit jak to myslim :D